docs/development/release-process.rst

```diff
@@ -0,0 +1,96 @@
+=====================
+pip's release process
+=====================
+
+
+Release Cadence
+===============
+
+The pip project has a release cadence of releasing whatever is on ``master``
+every 3 months. This gives users a predictable pattern for when releases
+are going to happen and prevents locking up improvements for fixes for long
+periods of time, while still preventing massively fracturing the user base
+with version numbers.
+
+Our release months are January, April, July, October. The release date within
+that month will be up to the release manager for that release. If there are
+no changes, then that release month is skipped and the next release will be
+3 month later.
+
+The release manager may, at their discretion, choose whether or not there
+will be a pre-release period for a release, and if there is may extend that
+period into the next month if needed.
+
+Because releases are made direct from the ``master`` branch, it is essential
+that ``master`` is always in a releasable state. It is acceptable to merge
+PRs that partially implement a new feature, but only if the partially
+implemented version is usable in that state (for example, with reduced
+functionality or disabled by default). In the case where a merged PR is found
+to need extra work before being released, the release manager always has the
+option to back out the partial change prior to a release. The PR can then be
+reworked and resubmitted for the next release.
+
+
+Deprecation Policy
+==================
+
+Any change to pip that removes or significantly alters user-visible behaviour
+that is described in the pip documentation will be deprecated for a minimum of
+one released version before the change occurs. Deprecation will take the form of
+a warning being issued by pip when the feature is used. Longer deprecation
+periods, or deprecation warnings for behaviour changes that would not normally
+be covered by this policy, are also possible depending on circumstances, but
+this is at the discretion of the pip developers.
+
+Note that the documentation is the sole reference for what counts as agreed
+behaviour. If something isn't explicitly mentioned in the documentation, it can
+be changed without warning, or any deprecation period, in a pip release.
+However, we are aware that the documentation isn't always complete - PRs that
+document existing behaviour with the intention of covering that behaviour with
+the above deprecation process are always acceptable, and will be considered on
+their merits.
+
+
+Release Ops
+===========
```

Comment:
I'm not keen on "Ops" - can this be retained as "Process"? Even expanding the abbreviation, "Release Operations" sounds weird to me.

Reply:
Sure. :)